ESP8266/ESP8266.cpp

```diff
@@ -122,6 +124,9 @@ nsapi_error_t ESP8266::connect(const char *ap, const char *passPhrase)
 {
     _smutex.lock();
     setTimeout(ESP8266_CONNECT_TIMEOUT);
+    _connection_status = NSAPI_STATUS_CONNECTING;
+    if(_connection_status_cb)
+        _connection_status_cb(NSAPI_EVENT_CONNECTION_STATUS_CHANGE, _connection_status);
```

Comment:
Should this be done before locking the mutex as we don't know what the callback function might be doing.

Reply:
Possible deadlock, I would prefer to call after the mutex is unlocked.

Reply:
This can be moved before the call to _smutex.lock(), in any case, the connected and disconnected states are still noticed from oob's. oob's are also called from a context where mutex is locked. Should the oob-processing and triggering callback from there be moved elsewhere?